patches/minecraft/net/minecraft/world/storage/MapDecoration.java.patch

```diff
@@ -0,0 +1,32 @@
+--- ../src-base/minecraft/net/minecraft/world/storage/MapDecoration.java
++++ ../src-work/minecraft/net/minecraft/world/storage/MapDecoration.java
+@@ -1,6 +1,8 @@
+ package net.minecraft.world.storage;
+ 
+ import net.minecraft.util.math.MathHelper;
++import net.minecraftforge.fml.common.FMLCommonHandler;
++import net.minecraftforge.fml.common.FMLLog;
+ import net.minecraftforge.fml.relauncher.Side;
+ import net.minecraftforge.fml.relauncher.SideOnly;
+ 
+@@ -121,6 +123,13 @@
+             this.field_191177_m = p_i47344_4_;
+         }
+ 
++        private Type(byte icon, boolean renderedOnFrame, int mapColor)
++        {
++            this.field_191175_k = icon;
++            this.field_191176_l = renderedOnFrame;
++            this.field_191177_m = mapColor;
++        }
++
+         public byte func_191163_a()
+         {
+             return this.field_191175_k;
+@@ -146,5 +155,6 @@
+         {
+             return values()[MathHelper.func_76125_a(p_191159_0_, 0, values().length - 1)];
+         }
++        public static MapDecoration.Type byName(String name) { for (MapDecoration.Type type : values()) { if (type.name().equals(name.toUpperCase())) return type; } FMLLog.warning("Tried to get an unknown MapDecoration.Type, returned default value! (" + name + ")"); return TARGET_X; }
```

Comment:
* don't cram it all into one line, it should at least be somewhat readable, if concise
* why are we calling toUppercase? What guarantees people only register uppercase names? I can register a name "foo" and this method will not work because it always tries to look for "FOO".
* Use `{}` instead of `+` in the logger lines